src/ethereum/arrow_glacier/fork.py

```diff
@@ -762,10 +763,13 @@ def process_transaction(
     sender = env.origin
     sender_account = get_account(env.state, sender)
 
-    if isinstance(tx, FeeMarketTransaction):
-        gas_fee = tx.gas * tx.max_fee_per_gas
-    else:
-        gas_fee = tx.gas * tx.gas_price
+    try:
+        if isinstance(tx, FeeMarketTransaction):
+            gas_fee = tx.gas * tx.max_fee_per_gas
+        else:
+            gas_fee = tx.gas * tx.gas_price
+    except ValueError as e:
```

Comment:
What raises `ValueError` here? Integer overflow?

If it is an integer overflow, we should handle this _for all_ integer overflows, and not just in these specific cases. Should we change `base_types` so that it throws a specific `OverflowError` that subclasses `ValueError`?

Reply:
> What raises `ValueError` here? Integer overflow?

Yes. This is due to overflows. This change has been made because the testing framework expects an `InvalidBlock` exception (as it should) but receives `ValueError`.


> If it is an integer overflow, we should handle this _for all_ integer overflows, and not just in these specific cases. Should we change `base_types` so that it throws a specific `OverflowError` that subclasses `ValueError`?

I did consider modifying the `base_types`, but raising an `InvalidBlock` exception (or a subclass of it) directly in the `base_types` does not seem right to me. (Although one could argue that the spec base types are used only in the context of processing blocks and so, an `InvalidBlock` exceptions is not out of place there). What do you think?

Alternatively, I also considered modifying the testing framework to handle the `ValueError` but that seems too generic of an exception to expect in the tests. Would be better to keep the expected exceptions as specific as possible.

Reply:
Another approach, which we take with gas tracking in the EVM, is to make everything a `Uint` instead of a `U256`. If we do that the overflow won't happen at all and instead `InvalidBlock` will get raised a few lines later.

Reply:
Updated to use `Uint` for all gas.